todo_app/app.py

```diff
@@ -40,13 +50,15 @@ def writer_required(f):
         @wraps(f)
         def decorated_function(*args, **kwargs):
             if get_user_role() == 'reader':
+                app.logger.info("user id %s tried to access an unauthorised page", flask_login.current_user.get_id())
```

Comment:
This should be a warn - there are two possible reasons this situation would arise:

- You have a bug where you presented a reader with a link that's only for writers
- Someone is going out of their way to try to access stuff they shouldn't.

Either way you will want to keep an eye on it and potentially do something about it.